chemprop/cli/train.py

```diff
@@ -686,8 +686,9 @@ def build_model(args, train_dset: MolGraphDataset | MulticomponentDataset) -> MP
         )
         model.bn.apply(lambda module: module.requires_grad_(False))
         for idx in range(args.frzn_ffn_layers):
-            model.predictor.ffn[idx * 3].requires_grad_(False)
-            setattr(model.predictor.ffn[idx * 3 + 2], "p", 0.0)
+            suffix = 0 if idx == 0 else 2
+            model.predictor.ffn[idx][suffix].requires_grad_(False)
+            setattr(model.predictor.ffn[idx + 1][1], "p", 0.0)
```

Comment:
I'm not sure if it's correct to freeze the subsequent dropout layer. I'm concerned that it might lead to overfitting for the task. For instance, consider a MLP with two blocks:
0 0 Linear
1 0 ReLU
1 1 Dropout
1 2 Linear

If this model is loaded, dropout layers are not used at all in the current approach. If it is used on a smaller dataset, it might easily become overfitted. In this case, having a dropout layer might be helpful.

Reply:
@davidegraff Do you have any thoughts on this?

Reply:
IMO dropout layers should be viewed in terms of what they're _preceding_ rather than what they're _succeeding_ when used as regularizers. So here, dropout is being used to regularize the inner layers by randomly masking out features in an attempt to suppress overfitting (i.e., decrease the layer's ability to rely on noise in the representation). The converse (viewing it as a regularizer of the preceding linear layer) doesn't really make sense to me because you can't zero out inputs of a layer that occurs before you.

Reply:
Thank you David for your insight. I appreciate the clarity on viewing dropout layers as regularizers for the layers that follow, rather than those that precede them. If I understand correctly, are we good to keep it as it is now?